darkcluster/src/main/java/com/linkedin/darkcluster/impl/DarkClusterStrategyFactoryImpl.java

```diff
@@ -109,18 +112,38 @@ public DarkClusterStrategy getOrCreate(@Nonnull String darkClusterName, @Nonnull
    */
   private DarkClusterStrategy createStrategy(String darkClusterName, DarkClusterConfig darkClusterConfig)
   {
-    if (darkClusterConfig.hasMultiplier() && darkClusterConfig.getMultiplier() > 0)
-    {
-      BaseDarkClusterDispatcher baseDarkClusterDispatcher = new BaseDarkClusterDispatcherImpl(darkClusterName, _darkClusterDispatcher,
-                                                                                              _notifier, _verifierManager);
-      return new RelativeTrafficMultiplierDarkClusterStrategy(_sourceClusterName, darkClusterName, darkClusterConfig.getMultiplier(),
-                                                              baseDarkClusterDispatcher, _notifier, _facilities.getClusterInfoProvider(), _random);
-    }
-    else
+    if (darkClusterConfig.hasMultiplierStrategyList())
     {
-      // Falling into this clause means that the user does not want traffic to be sent to the dark cluster, temporary or otherwise.
-      return new NoOpDarkClusterStrategy();
+      MultiplierStrategyTypeArray strategyList = darkClusterConfig.getMultiplierStrategyList();
+      for (com.linkedin.d2.multiplierStrategyType multiplierStrategyType : strategyList)
+      {
+        if (RELATIVE_TRAFFIC.equals(multiplierStrategyType) && darkClusterConfig.getMultiplier() > 0)
+        {
+          if (darkClusterConfig.getMultiplier() <= 0)
```

Comment:
this will never be reached because of previous "darkClusterConfig.getMultiplier() > 0"

Reply:
I think it would be a lot cleaner if we had the implementations themselves handle these edge cases rather than manually returning NoOpDarkClusterStrategy. In this case always return new RelativeTrafficMultiplierDarkClusterStrategy but since the multiplier is <= 0 then no requests will be sent out.

Reply:
yeah, sorry about that. I'll fix that code. The reasons I did not want to have the implementations handle it was because then every handleRequest invocation has to do an if check, rather than only when a strategy has to be created. The tradeoff of putting it in createStrategy seems to be worth not doing that if statement in the hot path.